java-checks/src/main/java/org/sonar/java/checks/EnumMutableFieldCheck.java

```diff
@@ -65,6 +63,24 @@ public void visitNode(Tree tree) {
       }
     }
   }
+  
+  private static boolean isNotStaticOrFinal(VariableTree variableTree) {
+    return !ModifiersUtils.hasModifier(variableTree.modifiers(), Modifier.STATIC)
+      && !ModifiersUtils.hasModifier(variableTree.modifiers(), Modifier.FINAL);
+  }
+  
+  private static boolean isMutableFinalMember(VariableTree variableTree) {
+    return ModifiersUtils.hasModifier(variableTree.modifiers(), Modifier.FINAL) && isMutableMember(variableTree);
+  }
+  
+  private static boolean isMutableMember(VariableTree variableTree) {
+    return variableTree.type().is(Tree.Kind.ARRAY_TYPE) || isDateOrCollection(variableTree);
+  }
+  
+  private static boolean isDateOrCollection(VariableTree variableTree) {
+    return variableTree.symbol().type().is("java.util.Date") || 
```

Comment:
extract the type as `variableTree.symbol().type()` in order to not have to call it 3 times in worst cases.